storage_controller/src/persistence.rs

```diff
@@ -1521,23 +1522,41 @@ impl Persistence {
         .await
     }
 
-    /// Load pending operations from db.
-    pub(crate) async fn list_pending_ops(
+    /// Load pending operations from db, joined together with timeline data.
+    pub(crate) async fn list_pending_ops_with_timelines(
         &self,
-    ) -> DatabaseResult<Vec<TimelinePendingOpPersistence>> {
+    ) -> DatabaseResult<Vec<(TimelinePendingOpPersistence, Option<TimelinePersistence>)>> {
```

Comment:
When would we return `None` for the second tuple element? I'd expect that a timeline operation only exists if the timeline itself exists.

Do we clean up when deleting a timeline/tenant?

Reply:
There is no foreign key constraint, so it's possible that there is pending ops in the db without a timeline entry in the db.

There is two options to handle such situations: either ignore them (via a inner join), or to pass it on and then warn about them in the logs.

We don't do specific cleanups, but good point, we should probably do it.

Reply:
You could also assert on it if we're confident.

Reply:
For now we can probably handle these cases manually, in the mid-long term we should probably do cleanup or something. Added it as todo